File: backend/src/main/java/com/synepth/kvdys/entity/Notification.java
package com.synepth.kvdys.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@Getter
@Setter
@NoArgsConstructor
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "recipient_id", nullable = false)
    private User recipient;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "actor_id")
    private User actor;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String message;

    private String targetUrl;

    @Column(nullable = false)
    private String type;

    @Column(nullable = false)
    private boolean isRead = false;

    @Column(nullable = false)
    private LocalDateTime createdAt;
}

File: frontend/src/app/models/notification.ts
export interface NotificationResponse {
  id: number;
  recipientId: number;
  actorId?: number | null;
  actorName?: string | null;
  actorUsername?: string | null;
  actorAvatarUrl?: string | null;
  title: string;
  message: string;
  targetUrl?: string | null;
  type: string;
  read?: boolean;
  isRead?: boolean;
  createdAt: string;
}

export interface UnreadCountResponse {
  unreadCount: number;
}
